src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/Logging/MSBuildLog.cs

```diff
@@ -8,8 +8,11 @@ namespace Microsoft.DotNet.ApiSymbolExtensions.Logging
     /// <summary>
     /// Class to define common logging abstraction for MSBuild tasks across the APICompat and GenAPI codebases.
     /// </summary>
-    internal class MSBuildLog(Logger log) : ILog
+    internal class MSBuildLog(Logger log, string? noWarn = null) : ILog
```

Comment:
Is there a reason why you opted for a `string?` instead of a `string[]?`?

I think the usage of ';' as a separator should not be a concern for the constructor caller.

Reply:
This is copied from https://github.com/dotnet/sdk/blob/1875146911221e683f0672b5a30aeaf0b288e3d6/src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs#L20

Reply:
Is the ';' necessary there?

Note: changes don't need to be made in this PR. Just something to consider for a separate change.

Reply:
> Is the ';' necessary there?

Not sure I fully understand your question. The NoWarn string usually looks like the following: `CS1591;PKV001;CP1002`. The delimiter splits the individual no warn parts.

Reply:
Ok so what I'm hearing is that usually the list of nowarns are extracted directly from an msbuild property, as opposed to having a hardcoded list directly in source code. Correct?

Reply:
APICompat doesn't have access to compiler directives like "pragma warning disable..." in source code. Its input is IL assemblies. To supress log warnings in APICompat, you need to use NoWarn which can be set as a property when using the msbuild task or as a CLI switch when using the global tool.

I hope that makes things clearer.

Reply:
Sort of. My question was about the way the tools retrieved the nowarns. Are they obtained from the csprojs? There's no case where we have them hardcoded in a field in a class in source code?

Anyway, not too important, we can resolve this.